dataset/dataset.cpp

```diff
@@ -19,126 +19,49 @@ template <class T> typename T::view_type makeViewItem(T &variable) {
     return typename T::view_type(variable);
 }
 
-namespace {
-constexpr auto contains = [](const auto &dims, const Dim dim) {
-  if constexpr (std::is_same_v<std::decay_t<decltype(dims)>, Dimensions>)
-    return dims.contains(dim);
-  else
-    return dims.count(dim) == 1;
-};
+template <class T1> auto makeViewItems(T1 &coords) {
+  std::unordered_map<typename T1::key_type, VariableView> items;
+  for (auto &item : coords)
+    items.emplace(item.first, makeViewItem(item.second));
+  return items;
 }
 
-template <class View, class Dims, class T1>
+template <class Dims, class T1>
 auto makeViewItems(const Dims &dims, T1 &coords) {
-  typename View::holder_type items;
-  for (auto &item : coords) {
-    // We preserve only items that are part of the space spanned by the
-    // provided parent dimensions.
-    auto contained = [&dims](const auto &item2) {
-      const auto &coordDims = item2.second.dims();
-      if constexpr (std::is_same_v<typename View::key_type, Dim>) {
-        const bool is_dimension_coord =
-            !contains_events(item2.second) && coordDims.contains(item2.first);
-        return coordDims.empty() || contains_events(item2.second) ||
-               (is_dimension_coord ? contains(dims, item2.first)
-                                   : contains(dims, coordDims.inner()));
-      } else
-        return coordDims.empty() || contains(dims, coordDims.inner());
-    };
-    if (contained(item)) {
+  std::unordered_map<typename T1::key_type, VariableView> items;
+  // We preserve only items that are part of the space spanned by the
+  // provided parent dimensions.
+  auto contained = [&dims](const auto &item) {
+    for (const Dim dim : item.second.dims().labels())
+      if (dims.count(dim) == 0)
```

Comment:
What's wrong with calling `dims.contains`? wouldn't that be more readable here? I haven't seen a usage of `count` yet that couldn't be covered with `contains`

Reply:
`std::unordered_map` does not have `contains` (it will in C++20). This is generic code that may be used for maps  as well as `Dimensions`.